CASSANDRA-17406: March 2022 blog "Apache Cassandra Changelog 13" (#105)
apache/cassandra-website: site-content/source/modules/ROOT/pages/blog/Apache-Cassandra-Changelog-13-March-2022.adoc

image::blog/changelog_header.jpg[Apache Cassandra Changelog]
Our monthly roundup of key activities and knowledge to keep the community informed.

== Release Notes

=== Released

The latest release of Apache Cassandra is https://www.apache.org/dyn/closer.lua/cassandra/4.0.3/apache-cassandra-4.0.3-bin.tar.gz[4.0.3^] (https://downloads.apache.org/cassandra/4.0.3/apache-cassandra-4.0.3-bin.tar.gz.asc[pgp^], https://downloads.apache.org/cassandra/4.0.3/apache-cassandra-4.0.3-bin.tar.gz.sha256[sha256^], and https://downloads.apache.org/cassandra/4.0.3/apache-cassandra-4.0.3-bin.tar.gz.sha512[sha512^]), which has been available since 17 February 2022. We released new versions of all supported versions of Cassandra (https://www.apache.org/dyn/closer.lua/cassandra/3.11.12/apache-cassandra-3.11.12-bin.tar.gz[3.11.12^], https://www.apache.org/dyn/closer.lua/cassandra/3.0.26/apache-cassandra-3.0.26-bin.tar.gz[3.0.26^]) to address a vulnerability https://lists.apache.org/thread/y4nb9s4co34j8hdfmrshyl09lokm7356[CVE-2021-44521^].

Essentially, if you're running Cassandra in the following non-default configuration, below, it's possible for an attacker to execute arbitrary code on the host:

> * `enable_user_defined_functions: true`

Reviewer: I'd suggest formatting as code:
````suggestion
```
enable_user_defined_functions: true
````

> * `enable_scripted_user_defined_functions: true`

Reviewer: ```suggestion
enable_scripted_user_defined_functions: true
```

> * `enable_user_defined_functions_threads: false`

Reviewer: ````suggestion
enable_user_defined_functions_threads: false
```
````

The attacker also needs permission to create user-defined functions as well as this configuration arrangement.

We suggest 3.0 users should upgrade to 3.0.26; 3.11 users should upgrade to 3.11.12; and 4.0 users should upgrade to 4.0.3.

Thanks to Omer Kaspi of the JFrog Security vulnerability research team for the discovery.
 

> Please read the https://gitbox.apache.org/repos/asf?p=cassandra.git;a=blob_plain;f=NEWS.txt;hb=refs/tags/cassandra-4.0.3[release notes^] and https://issues.apache.org/jira/browse/CASSANDRA[let us know^] if you encounter any problems.

Reviewer: Unfortunately, the release notes link doesn't contain the info about the CVE so we need to replace it:

```suggestion
Please read the https://github.com/apache/cassandra/blob/cassandra-4.0/NEWS.txt[release notes^] and https://issues.apache.org/jira/browse/CASSANDRA[let us know^] if you encounter any problems.
```

== User Space

=== Kinetic Data

Kinetic Data developed a low-code system, a forms and workflow engine built on top of Apache Cassandra, where, for example, users can define a form with drag and drop fields and store the data in Cassandra.

"Once it's set up and running it’s hands off. Quite frankly, it's easy from an operations perspective [...] so our customers, they're using Cassandra, but they don't really realize it [...]. But they do say, ‘it's always up. It's always fast.’ It's all these benefits that you really want the end-user to know about."
-- Joe Sundberg, CEO of Kinetic Data

_Do you have a Cassandra case study to share? Email cassandra@constantia.io._

== In the News

The New Stack: 
https://thenewstack.io/jfrog-finds-rce-issue-in-apache-cassandra/[JFrog Finds RCE Issue in Apache Cassandra^]

== Cassandra Tutorials & More

xref:blog/Apache-Cassandra-and-Java-SE-11-support.adoc[Apache Cassandra and Java SE 11] - Chris Thornett

> xref:Behind-the-scenes-of-an-Apache-Cassandra-Release.adoc[Behind the Scenes of an Apache Cassandra Release] - Josh McKenzie

Reviewer: xref missing leading `blog/`:

```suggestion
xref:blog/Behind-the-scenes-of-an-Apache-Cassandra-Release.adoc[Behind the Scenes of an Apache Cassandra Release] - Josh McKenzie
```
